dali/pipeline/data/tensor.h

```diff
@@ -353,10 +356,12 @@ class Tensor : public Buffer<Backend> {
    * of a Tensor
    */
   inline void Squeeze() {
-    shape_.erase(std::remove(shape_.begin(), shape_.end(), 1), shape_.end());
-    if (shape_.empty()) {
-      shape_.push_back(1);
+    std::vector<Index> shape(shape_.begin(), shape_.end());
```

Comment:
Why it is needed?

Reply:
Because we don't support things like erase and push_back in TensorShape<> and I didn't want to reimplement this part by hand.